stan/math/prim/prob/neg_binomial_lpmf.hpp

```diff
@@ -65,69 +65,46 @@ return_type_t<T_shape, T_inv_scale> neg_binomial_lpmf(const T_n& n,
     }
   }
 
-  VectorBuilder<true, T_partials_return, T_inv_scale> log_beta(size_beta);
+  VectorBuilder<true, T_partials_return, T_inv_scale> log1p_inv_beta(size_beta);
   VectorBuilder<true, T_partials_return, T_inv_scale> log1p_beta(size_beta);
-  VectorBuilder<true, T_partials_return, T_inv_scale> log_beta_m_log1p_beta(
-      size_beta);
   for (size_t i = 0; i < size_beta; ++i) {
     const T_partials_return beta_dbl = value_of(beta_vec[i]);
-    log_beta[i] = log(beta_dbl);
+    log1p_inv_beta[i] = log1p(inv(beta_dbl));
     log1p_beta[i] = log1p(beta_dbl);
-    log_beta_m_log1p_beta[i] = log_beta[i] - log1p_beta[i];
   }
 
-  VectorBuilder<true, T_partials_return, T_shape, T_inv_scale> lambda(
-      size_alpha_beta);
-  VectorBuilder<true, T_partials_return, T_shape, T_inv_scale>
-      alpha_log_beta_over_1p_beta(size_alpha_beta);
   VectorBuilder<!is_constant_all<T_inv_scale>::value, T_partials_return,
                 T_shape, T_inv_scale>
       lambda_m_alpha_over_1p_beta(size_alpha_beta);
-  for (size_t i = 0; i < size_alpha_beta; ++i) {
-    const T_partials_return alpha_dbl = value_of(alpha_vec[i]);
-    const T_partials_return beta_dbl = value_of(beta_vec[i]);
-    lambda[i] = alpha_dbl / beta_dbl;
-    alpha_log_beta_over_1p_beta[i] = alpha_dbl * log_beta_m_log1p_beta[i];
-    if (!is_constant_all<T_inv_scale>::value) {
-      lambda_m_alpha_over_1p_beta[i] = lambda[i] - alpha_dbl / (1 + beta_dbl);
+  if (!is_constant_all<T_inv_scale>::value) {
```

Comment:
Given that this check and loop already exists below, I'd move these calculations there. Otherwise there'd be the cost associated with traversing large inputs twice

Reply:
Similarly with the calculations of ```digamma_alpha``` above

Reply:
It's possible that alpha and beta are both size 1 but y is length N > 1, so this is faster possibly.

Reply:
Ah, I missed that part, cool beans